edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -20,25 +20,32 @@ public sealed class DeviceScopeIdentitiesCache : IDeviceScopeIdentitiesCache
         readonly IKeyValueStore<string, string> encryptedStore;
         readonly IServiceIdentityHierarchy serviceIdentityHierarchy;
         readonly Timer refreshCacheTimer;
-        readonly TimeSpan refreshRate;
+        readonly TimeSpan periodicRefreshRate;
+        readonly TimeSpan refreshDelay;
         readonly AsyncAutoResetEvent refreshCacheSignal = new AsyncAutoResetEvent();
         readonly AsyncManualResetEvent refreshCacheCompleteSignal = new AsyncManualResetEvent();
         readonly object refreshCacheLock = new object();
 
         Task refreshCacheTask;
+        DateTime cacheLastRefreshTime;
+        Dictionary<string, DateTime> identitiesLastRefreshTime;
 
         DeviceScopeIdentitiesCache(
             IServiceIdentityHierarchy serviceIdentityHierarchy,
             IServiceProxy serviceProxy,
             IKeyValueStore<string, string> encryptedStorage,
             IDictionary<string, StoredServiceIdentity> initialCache,
-            TimeSpan refreshRate)
+            TimeSpan periodicRefreshRate,
+            TimeSpan refreshDelay)
         {
             this.serviceIdentityHierarchy = serviceIdentityHierarchy;
             this.serviceProxy = serviceProxy;
             this.encryptedStore = encryptedStorage;
-            this.refreshRate = refreshRate;
-            this.refreshCacheTimer = new Timer(this.RefreshCache, null, TimeSpan.Zero, refreshRate);
+            this.periodicRefreshRate = periodicRefreshRate;
+            this.refreshDelay = refreshDelay;
+            this.refreshCacheTimer = new Timer(this.RefreshCache, null, TimeSpan.Zero, this.periodicRefreshRate);
```

Comment:
nit: you may use PeriodicTask instead of Timer.

Reply:
this is how it was before so I'll leave it.